docs/DafnyRef/Expressions.md

```diff
@@ -52,46 +52,31 @@ in order of increasing binding power.
  `/`                      | 7 | division (divided by)
  `%`                      | 7 | modulus (mod)
 --------------------------|------------------------------------
- `|`                      | 8 | bit-wise or
- `&`                      | 8 | bit-wise and
- `^`                      | 8 | bit-wise exclusive-or (not equal)
+ `|`                      | 8 | [bit-wise or](#sec-bitvector-expression)
+ `&`                      | 8 | [bit-wise and](#sec-bitvector-expression)
+ `^`                      | 8 | [bit-wise exclusive-or (not equal)](#sec-bitvector-expression)
 --------------------------|------------------------------------
- `as` operation           | 9 | [type conversion](#sec-as-expression)
- `is` operation           | 9 | [type test](#sec-as-expression)
+ `as` operation           | 9 | [type conversion](#sec-as-is-expression)
+ `is` operation           | 9 | [type test](#sec-as-is-expression)
 --------------------------|------------------------------------
  `-`                      | 10 | arithmetic negation (unary minus)
  `!`                      | 10 | logical negation, bit-wise complement
 --------------------------|------------------------------------
  Primary Expressions      | 11 |
 
-We are calling the ``UnaryExpression``s that are neither
-arithmetic nor logical negation the _primary expressions_.
-They are the most tightly bound.
-
-In the grammar entries below we explain the meaning when the
-operator for that precedence level is present. If the
-operator is not present then we just descend to the
-next precedence level.
-
-## 21.1. Top-level expressions {#sec-top-level-expression}
-````grammar
-Expression(allowLemma, allowLambda) =
-    EquivExpression(allowLemma, allowLambda)
-    [ ";" Expression(allowLemma, allowLambda) ]
-````
-
-The "allowLemma" argument says whether or not the expression
-to be parsed is allowed to have the form `S;E` where `S` is a call to a lemma.
-"allowLemma" should be passed in as "false" whenever the expression to
-be parsed sits in a context that itself is terminated by a semi-colon.
-
-The "allowLambda" says whether or not the expression to be parsed is
-allowed to be a lambda expression.  More precisely, an identifier or
-parenthesized-enclosed comma-delimited list of identifiers is allowed to
-continue as a lambda expression (that is, continue with a `reads`, `requires`,
-or `=>`) only if "allowLambda" is true.  This affects function/method/iterator
-specifications, if/while statements with guarded alternatives, and expressions
-in the specification of a lambda expression itself.
+
+## 21.1. Top-level expressions ([grammar](#g-top-level-expression)) {#sec-top-level-expression}
+
+Examples:
+<!-- %no-check -->
+```dafny
+let a == 2*b; a*b
```

Comment:
Is this valid Dafny syntax? I don't seem to be able to find any example of it, or mention in the grammar.